config/config.go

```diff
@@ -144,6 +145,72 @@ func (sc *SafeConfig) ReloadConfig(confFile string, logger *slog.Logger) (err er
 	return nil
 }
 
+// CELProgram encapsulates a cel.Program and makes it YAML marshalable.
+type CELProgram struct {
+	cel.Program
+	Expression string
+}
+
+// NewCELProgram creates a new CEL Program and returns an error if the
+// passed-in CEL expression does not compile.
+func NewCELProgram(s string) (CELProgram, error) {
+	program := CELProgram{
+		Expression: s,
+	}
+
+	env, err := cel.NewEnv(
+		cel.Variable("body", cel.MapType(cel.StringType, cel.DynType)),
```

Comment:
To allow any type, this can be declared as a totally dynamic type:

```suggestion
		cel.Variable("body", cel.DynType),
```